### ISSUE TITLE
Refactor and deprecate

### ISSUE THREAD

**Issue by Stebalien** [libp2p/go-addr-util#18]
Looking at this package:

1. It's _only_ used from go-libp2p-swarm.
2. Much of it's functionality belongs in either swarm or go-multiaddr-net (where it will be more discoverable).

In an effort to slim down our package list a bit, we should copy any functions we need into their respective places and then deprecate (BUT NOT REMOVE) this package.

**Comment by Stebalien** [comment]
@anacrolix interested?

**Comment by anacrolix** [comment]
It seems like a good idea. I see https://godoc.org/github.com/libp2p/go-addr-util?importers. We could definitely archive this repo to ensure nothing more gets added.

**Comment by anacrolix** [comment]
It feels like busy work, unless something in here is wrong, or may mislead users.

**Comment by Stebalien** [comment]
No, this is super-low priority. I noticed because I was fixing a bug in this library and realized we mostly don't need it.

**Comment by Stebalien** [comment]
(I just know you're interested in reducing our repo count).

**Comment by ghost** [comment]
I can take this as part of multiformats/multiaddr#88

**Comment by raulk** [comment]
Agree with this change. Too much unnecessary indirection at the moment.
